test/common/run-tests

```diff
@@ -356,8 +358,9 @@ def detect_tests(test_files, image, opts):
                 rwa = not testlib.get_decorator(test_method, test, "no_retry_when_changed")
                 todo = testlib.get_decorator(test_method, test, "todo")
                 if getattr(test.__class__, "provision", None):
-                    # each additionally provisioned VM costs destructive test capacity
-                    cost = len(test.__class__.provision)
+                    # each additionally provisioned VM memory costs destructive test capacity
+                    total_memory = sum(config.get('memory_mb', 0) for _, config in test.__class__.provision.items())
```

Comment:
The `.get()` default seems wrong -- surely it should be `DEFAULT_MACHINE_MEMORY_MB`, not `0`?

You can [look at the existing test](https://cockpit-logs.us-east-1.linodeobjects.com/pull-22163-9faf1b68-20250703-121700-rhel-10-1-expensive/log.html#71) and see that the cost is 2, but it should be 3.